ironmon_tracker/Battle.lua

```diff
@@ -284,131 +361,89 @@ function Battle.checkEnemyEncounter(opposingPokemon)
 	end
 end
 
--- Check if the opposing Pokemon used a move (it's missing pp from max), and if so track it
-function Battle.checkEnemyMovesUsed(opposingPokemon)
-	if Battle.enemyTransformed then return end
+function Battle.checkAbilitiesToTrack()
 
-	for _, move in pairs(opposingPokemon.moves) do
-		if MoveData.isValid(move.id) then
-			local moveUsed = false
-
-			-- Manually track Focus Punch, since PP isn't deducted if the mon charges the move but then dies
-			if move.id == 264 and Battle.attacker % 2 == 1 and Battle.battleMsg ~= 0 and Battle.battleMsg == GameSettings.BattleScript_FocusPunchSetUp then
-				moveUsed = true
-			elseif move.pp < tonumber(MoveData.Moves[move.id].pp) then
-				moveUsed = true
-			end
-
-			if moveUsed then
-				Tracker.TrackMove(opposingPokemon.pokemonID, move.id, opposingPokemon.level)
-
-				if move.id == 144 then -- 144 = Transform
-					Battle.enemyTransformed = true
-				end
-			end
-		end
-	end
-end
-
--- Checks if ability should be auto-tracked. Returns true if so; false otherwise
-function Battle.checkEnemyAbilityUsed(enemyAbility, playerAbility)
+	-- Track previous ability activation for handling Synchronize
 	if Battle.Synchronize.turnCount < Battle.turnCount then
 		Battle.Synchronize.turnCount = Battle.turnCount
 		Battle.Synchronize.battler = -1
 		Battle.Synchronize.attacker = -1
 		Battle.Synchronize.battlerTarget = -1
 	end
-
-	--Levitate doesn't get a message, so it gets checked independently
-	if enemyAbility == 26 and Battle.attacker % 2 == 0 then
-		local levitateCheck = Memory.readbyte(GameSettings.gBattleCommunication + 0x6)
-		if levitateCheck == 4 then
-		-- Requires checking gBattleCommunication for the B_MSG_GROUND_MISS flag (4)
-			return true
+	local indexesToTrack = {}
```

Comment:
what are "indexes" that need to be tracked? Is this "battle index for the pokemon combatants?"

Reply:
Yes, the return value will be an array of values 0 to 3 inclusive, representing the index of the combatant in the battle (0 amd 2 are allies, 1 and 3 are the enemies).
The ability tracker previously returned true or false to indicate whether the opponent's pokemon's ability should be tracked. Now, it determines the specific pokemon whose ability needs to be tracked. In cases where one ability directly triggers another (i.e. Intimidate vs Clear Body), this allows both pokemon's abilities to be tracked from the one message.